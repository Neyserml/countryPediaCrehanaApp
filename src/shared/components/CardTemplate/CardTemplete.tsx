import React, { ReactElement, ReactNode } from "react";
import styled from "styled-components";

const StyledImg = styled.img`
  height: 100%;
  width: 15%;
  max-height: 100px;
  min-height: 40px;
`;

const StyledWrapper = styled.div`
  height: 300px;
`;

export interface CardTemplateInterface {
  img: string;
  title: string;
  children: ReactNode;
  onClickDetail: (title: string) => void;
}

const CardTemplate = ({
  img,
  title,
  children,
  onClickDetail,
}: CardTemplateInterface): ReactElement => {
  const wiewName = (): void => {
    onClickDetail(title);
  };

  return (
    <StyledWrapper>
      <div onClick={wiewName} className="card">
        <div className="card-image waves-effect waves-block waves-light">
          <StyledImg src={img} />
        </div>
        <div className="card-content">
          <span className="card-title activator grey-text text-darken-4">
            <b>{title}</b>
          </span>
          {children}
        </div>
      </div>
    </StyledWrapper>
  );
};

export default CardTemplate;
