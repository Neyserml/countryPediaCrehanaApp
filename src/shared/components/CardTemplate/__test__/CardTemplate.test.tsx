import { render } from "@testing-library/react";
import React, { ReactNode } from "react";
import CardTemplate from "../CardTemplete";

const renderComponent = (img: string, title: string, children: ReactNode) => {
  return render(
    <CardTemplate img={img} title={title} onClickDetail={jest.fn()}>
      {children}
    </CardTemplate>
  );
};

describe("Card Template Component", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("should match snapshot in card template component", () => {
    const children = <h3>prueba</h3>;
    const component = renderComponent("img.jpg", "Peru", children);
    expect(component.container).toMatchSnapshot();
  });
});
