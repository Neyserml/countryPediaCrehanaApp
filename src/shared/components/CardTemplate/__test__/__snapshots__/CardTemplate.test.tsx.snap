// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Template Component should match snapshot in card template component 1`] = `
<div>
  <div
    class="sc-gsTCUz fwTMXJ"
  >
    <div
      class="card"
    >
      <div
        class="card-image waves-effect waves-block waves-light"
      >
        <img
          class="sc-bdfBwQ iPjGkV"
          src="img.jpg"
        />
      </div>
      <div
        class="card-content"
      >
        <span
          class="card-title activator grey-text text-darken-4"
        >
          <b>
            Peru
          </b>
        </span>
        <h3>
          prueba
        </h3>
      </div>
    </div>
  </div>
</div>
`;
